boot-qemu.py

```diff
@@ -167,8 +199,38 @@ def _run_gdb(self):
             if answer.lower() == 'n':
                 break
 
-    def run(self):
-        # Make sure QEMU binary is configured and available
+    def _set_kernel_vars(self):
+        if self.kernel:
+            if not self.kernel_dir:
+                self.kernel_dir = self.kernel.parent
+            # Nothing else to do, kernel image and build folder located and set
+            return
+
+        if not self.kernel_dir:
+            raise RuntimeError(
+                'No kernel image or kernel build folder specified?')
+        if not self._default_kernel_path:
+            raise RuntimeError('No default kernel path specified?')
+
+        possible_kernel_locations = {
+            Path(self.kernel_dir,
+                 self._default_kernel_path),  # default (kbuild)
+            Path(self.kernel_dir, self._default_kernel_path.name),  # tuxmake
+        }
+        for loc in possible_kernel_locations:
+            if loc.exists():
+                self.kernel = loc
+                break
```

Comment:
`utils.find_first_file`?

Reply:
Yes, much cleaner that way :)

fc98ddcb9651ca1c1f6a367b56973100fbd669d9